src/http/one/Parser.cc

```diff
@@ -102,7 +112,7 @@ LineCharacters()
 void
 Http::One::Parser::cleanMimePrefix()
 {
-    Http1::Tokenizer tok(mimeHeaderBlock_);
+    ::Parser::Tokenizer tok(mimeHeaderBlock_);
```

Comment:
Done.

Reply:
This change may need more work. The (implemented) second approach was about removing the namespace prefix in most cases, but I still see `Parser::` prefix inside the `Parser` class. Can that be removed? As a general rule of thumb, we should not say `Foo::bar` inside `Foo`.

Reply:
Ok, removed Parser:: prefix from the Parser class and its children. Firstly I thought that we need only to get rid only of '::'.

Reply:
Please be very careful with this removal. The ::Parser needs to be used explicitly when the generic syntax rules are being applied instead of the HTTP/1.x-specific rules (Http1::Parser). Early HTTP RFCs had some conflicting syntax requirements about tolerances being allowed or forbidden depending on the token placement in the HTTP message and sometimes even what HTTP extension header was being parsed.

Without the namespace scoping it is particularly hard to tell from any given code function which Parser is controlling the applied rules. I expect we will end up with people doing things like what you did with this PR initially: "fixing" to tolerateRFC 2616-style behaviour unknowing that the code is also handling RFC7230 coinstructs where tolerances are expressly forbidden, then changing it back, repeating indefinitely.

FWIW: HTTP/2 extends the problem further with a zero-tolerance requirement for parts of the protocol explicitly defined and inheriting BWS/RWS/OWS tolerance in other extension parts. The Http1::Parser class was existing as part of a pair with Http2::Parser class under construction to separate those distinct trees of parsing rules such that derivative Parser for other protocols could inherit the right set of rules without us needing duplicate the logic in each protocol Parser.

Reply:
Since Http::One::Tokenizer was removed in this PR there should be no confusion: all Http::One code now uses the generic ::Parser::Tokenizer, which declared as a public typedef in Http::One::Parser.  In other words, there should be no need in duplicating "::Parser::" prefix in dozens of Http::One places, since there is a single generic Tokenizer there.